.travis.yml

```
@@ -1,59 +1,60 @@
-      else
-          wget "http://repo.continuum.io/miniconda/Miniconda2-$MINCONDA_VERSION-$MINCONDA_OS.sh" -O miniconda.sh;
-      fi
-    - bash miniconda.sh -b -p $HOME/miniconda
-    - export PATH="$HOME/miniconda/bin:$PATH"
-    - hash -r
-    - conda config --set always_yes yes --set changeps1 no
-    - conda update -q conda
-    - conda install cmake nlohmann_json=3.2.0 pybind11=2.2.4 -c conda-forge
-    - if [[ "$TRAVIS_OS_NAME" == "linux" ]]; then
-        conda install gxx_linux-64 libgcc -c conda-forge;
-      fi
-    # Activate root environment
-    - source activate root
-    - mkdir build
-    - cd build
-    - cmake -D CMAKE_INSTALL_PREFIX=$HOME/miniconda -D DOWNLOAD_GTEST=ON -D PYTHON_EXECUTABLE=$HOME/miniconda/bin/python -D CMAKE_INSTALL_LIBDIR=lib -DCMAKE_C_COMPILER=$CC -DCMAKE_CXX_COMPILER=$CXX ..
-    - make -j2 install
-script:
-    - ./test/test_pybind11_json
+language: cpp
+env:
+  global:
+    - MINCONDA_VERSION="latest"
+    - MINCONDA_LINUX="Linux-x86_64"
+    - MINCONDA_OSX="MacOSX-x86_64"
+matrix:
+  fast_finish: true
+  include:
+    - os: linux
+      addons:
+        apt:
+          sources:
+            - ubuntu-toolchain-r-test
+      env: PY=3
+    - os: linux
+      addons:
+        apt:
+          sources:
+            - ubuntu-toolchain-r-test
+      env: PY=2
+    - os: osx
+      osx_image: xcode8
+      compiler: clang
+      env: PY=3
+before_install:
+    - |
+      if [[ "$TRAVIS_OS_NAME" == "osx" ]]; then
+        export CXX=clang++ CC=clang PYTHONHOME=$HOME/miniconda;
+      fi
+install:
+    # Define the version of miniconda to download
+    - if [[ "$TRAVIS_OS_NAME" == "linux" ]]; then
+        MINCONDA_OS=$MINCONDA_LINUX;
+      elif [[ "$TRAVIS_OS_NAME" == "osx" ]]; then
+        MINCONDA_OS=$MINCONDA_OSX;
+      fi
+    - if [[ "$PY" == "3" ]]; then
+          wget "http://repo.continuum.io/miniconda/Miniconda3-$MINCONDA_VERSION-$MINCONDA_OS.sh" -O miniconda.sh;
+      else
+          wget "http://repo.continuum.io/miniconda/Miniconda2-$MINCONDA_VERSION-$MINCONDA_OS.sh" -O miniconda.sh;
+      fi
+    - bash miniconda.sh -b -p $HOME/miniconda
+    - export PATH="$HOME/miniconda/bin:$PATH"
+    - hash -r
+    - conda config --set always_yes yes --set changeps1 no
+    - conda update -q conda
+    - conda install cmake nlohmann_json=3.2.0 pybind11=2.2.4 -c conda-forge
+    - if [[ "$TRAVIS_OS_NAME" == "linux" ]]; then
+        conda install gxx_linux-64 libgcc -c conda-forge;
+      fi
+    # Activate root environment
+    - source activate root
+    - mkdir build
+    - cd build
+    - cmake -D CMAKE_INSTALL_PREFIX=$HOME/miniconda -D DOWNLOAD_GTEST=ON -D PYTHON_EXECUTABLE=$HOME/miniconda/bin/python -D CMAKE_INSTALL_LIBDIR=lib -DCMAKE_C_COMPILER=$CC -DCMAKE_CXX_COMPILER=$CXX ..
+    - make -j2 install
+script:
+    - ./test/test_pybind11_json
+    - python ./test/test_pybind11_json.py

```

This is the cause of the CI failure, it breaks because you are in the `build` directory at this point.
So this would fix it:
```suggestion
    - cd ..
    - python ./test/test_pybind11_json.py 
```

---

thank you, I think my storage on onedrive may cause the error of diff.  Thank you for fixing CI. I would redo the fork, commit. to keep history soon